### PR TITLE
On collide attack

### PR DESCRIPTION
**Pull request by Mylab6**

### CODE REVIEW

File: Assets/Scripts/Enemies/AIMovement.cs [hunk 1]
```diff
@@ -49,20 +116,29 @@ void FixedUpdate()
             currentDirection = 1f;
         }
         //Store user input as a movement vector
```
Comment (townofdon) [inline Assets/Scripts/Enemies/AIMovement.cs:118]: Nit: this is not user input, right? Can we remove this comment to avoid confusion?

Comment (Mylab6) [reply Assets/Scripts/Enemies/AIMovement.cs:118]: Sure

File: Assets/Scripts/Enemies/EnemyAttack.cs [hunk 2]
```diff
@@ -6,9 +6,12 @@ public enum AimType
 {
     Player,
     Direction,
+    SingleCollision,
+    MultipleCollision
```
Comment (townofdon) [inline Assets/Scripts/Enemies/EnemyAttack.cs:10]: Could we be more descriptive here? I'm not sure what "Single Collision" or "Multiple Collision" means. Adding a code comment would also suffice to describe what's going on.

File: Assets/Scripts/Enemies/EnemyAttack.cs [hunk 3]
```diff
@@ -112,6 +124,12 @@ private void MeleeAttack()
         {
             return;
         }
+        if (aimType == AimType.SingleCollision)
+        {
+            canMelee = false;
```
Comment (townofdon) [inline Assets/Scripts/Enemies/EnemyAttack.cs:129]: Would there ever be a scenario to turn `canMelee` back to `true`?

**Pull request merged**